pkg/commands/fmt.go

```diff
@@ -0,0 +1,159 @@
+
+func newFmtCommand(logger logutils.Log, info BuildInfo) *fmtCommand {
+	c := &fmtCommand{
+		viper:     viper.New(),
+		log:       logger,
+		debugf:    logutils.Debug(logutils.DebugKeyExec),
+		cfg:       config.NewDefault(),
+		buildInfo: info,
+	}
+
+	fmtCmd := &cobra.Command{
+		Use:               "fmt",
+		Short:             "Format Go source files",
+		RunE:              c.execute,
+		PreRunE:           c.preRunE,
+		PersistentPreRunE: c.persistentPreRunE,
+		SilenceUsage:      true,
+	}
+
+	fmtCmd.SetOut(logutils.StdOut) // use custom output to properly color it in Windows terminals
+	fmtCmd.SetErr(logutils.StdErr)
+
+	flagSet := fmtCmd.Flags()
+	flagSet.SortFlags = false // sort them as they are defined here
+
+	setupConfigFileFlagSet(flagSet, &c.opts)
+
+	setupFormattersFlagSet(c.viper, flagSet)
+
+	c.cmd = fmtCmd
+
+	return c
+}
+
+func (c *fmtCommand) persistentPreRunE(cmd *cobra.Command, args []string) error {
+	c.log.Infof("%s", c.buildInfo.String())
+
+	loader := config.NewLoader(c.log.Child(logutils.DebugKeyConfigReader), c.viper, cmd.Flags(), c.opts, c.cfg, args)
+
+	err := loader.Load(config.LoadOptions{CheckDeprecation: true, Validation: true})
+	if err != nil {
+		return fmt.Errorf("can't load config: %w", err)
+	}
+
+	return nil
+}
+
+func (c *fmtCommand) preRunE(_ *cobra.Command, _ []string) error {
+	metaFormatter, err := goformatters.NewMetaFormatter(c.log, &c.cfg.Formatters, &c.cfg.Run)
+	if err != nil {
+		return fmt.Errorf("failed to create meta-formatter: %w", err)
+	}
+
+	matcher := processors.NewGeneratedFileMatcher(c.cfg.Formatters.Exclusions.Generated)
+
+	opts, err := goformat.NewRunnerOptions(c.cfg)
+	if err != nil {
+		return fmt.Errorf("build walk options: %w", err)
+	}
+
+	c.runner = goformat.NewRunner(c.log, metaFormatter, matcher, opts)
+
+	return nil
+}
+
+func (c *fmtCommand) execute(_ *cobra.Command, args []string) error {
+	if !logutils.HaveDebugTag(logutils.DebugKeyLintersOutput) {
+		// Don't allow linters and loader to print anything
+		log.SetOutput(io.Discard)
+		savedStdout, savedStderr := c.setOutputToDevNull()
+		defer func() {
+			os.Stdout, os.Stderr = savedStdout, savedStderr
+		}()
+	}
+
+	paths, err := cleanArgs(args)
+	if err != nil {
+		return fmt.Errorf("failed to clean arguments: %w", err)
+	}
+
+	c.log.Infof("Formatting Go files...")
+
+	err = c.runner.Run(paths)
+	if err != nil {
+		return fmt.Errorf("failed to process files: %w", err)
+	}
+
+	return nil
+}
+
+func (c *fmtCommand) setOutputToDevNull() (savedStdout, savedStderr *os.File) {
+	savedStdout, savedStderr = os.Stdout, os.Stderr
+	devNull, err := os.Open(os.DevNull)
+	if err != nil {
+		c.log.Warnf("Can't open null device %q: %s", os.DevNull, err)
+		return
+	}
+
+	os.Stdout, os.Stderr = devNull, devNull
+	return
+}
+
+func cleanArgs(args []string) ([]string, error) {
+	if len(args) == 0 {
+		abs, err := filepath.Abs(".")
+		if err != nil {
+			return nil, err
+		}
+
+		return []string{abs}, nil
+	}
+
+	var expended []string
```

Comment:
Perhaps here is a typo:
```suggestion
	var expanded []string
```